citro3d/src/uniform.rs

```diff
@@ -20,38 +22,147 @@ impl From<Index> for i32 {
     }
 }
 
-mod private {
-    use crate::math::Matrix;
+/// A uniform which may be bound as input to a shader program
+pub enum Uniform {
```

Comment:
This should probably be `#[non_exhaustive]` just in case we ever need to add a new enum variant for some reason

Reply:
Yeah, I guess strict backwards compatibility isn't a huge goal right now anyway but just feels safer to make these easier to extend in the future